### PR TITLE
Feature/idl 4.2 [4974]

### PR DESCRIPTION
**Pull request by LuisGP**
Changes to allow IDL 4.2.

### CODE REVIEW

File: src/it/java/com/eprosima/integration/TestManager.java [hunk 1]
```diff
@@ -46,6 +47,19 @@ public TestManager(TestLevel level, String generatorName, String inputPath, Stri
         this.generatorName = generatorName;
         this.inputPath = inputPath;
         this.outputPath = outputPath;
+        this.exampleArch = null;
```
Comment (richiware) [inline src/it/java/com/eprosima/integration/TestManager.java]: Why exampleArch? What is its purpose?

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.interp [hunk 2]
```diff
@@ -0,0 +1,414 @@
+token literal names:
```
Comment (richiware) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.interp]: For what is this new file?

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.tokens [hunk 3]
```diff
@@ -0,0 +1,223 @@
+T__0=1
```
Comment (richiware) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.tokens]: For what is this new file?

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.java [hunk 4]
```diff
@@ -0,0 +1,610 @@
+// Generated from /home/lgasco/git/eProsima/Fast-RTPS/IDL_4.2/thirdparty/idl/src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4 by ANTLR 4.7.1
```
Comment (richiware) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.java]: What is this new file for?

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.interp [hunk 5]
```diff
@@ -0,0 +1,383 @@
+token literal names:
```
Comment (richiware) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.interp]: What is this new file for?

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLParser.java [hunk 6]
```diff
@@ -0,0 +1,12241 @@
+// Generated from /home/lgasco/git/eProsima/Fast-RTPS/IDL_4.2/thirdparty/idl/src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4 by ANTLR 4.7.1
```
Comment (richiware) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLParser.java]: What is this new file for?

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4 [hunk 7]
```diff
@@ -1227,69 +1504,129 @@ member_def returns [Vector<Pair<Pair<String, Token>, Member>> ret = null]
...
         {
-           if($type_spec.typecode!=null)
-           {
-               for(int count = 0; count < $declarators.ret.size(); ++count)
-               {
-                   Member member = null;
+            /*
+            if (structTP != null)
+            {
+                System.out.println("Pertenezco a " + structTP.getName());
```
Comment (richiware) [inline src/main/antlr4/omg/com/eprosima/idl/parser/grammar/IDL.g4]: If this debug print is not more used, remove it :)

File: src/it/java/com/eprosima/integration/TestManager.java (hunk 1 above)
Comment (LuisGP) [reply src/it/java/com/eprosima/integration/TestManager.java]: The TestManager being used by XRCE was calling Micro-XRCE-Gen always with the exampleArch argument (architecture for generated examples), but Fast-RTPS-Gen doesn't need it for testing, so I added the possibility to call the TestManager without the exampleArch argument.

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.interp (hunk 2 above)
Comment (LuisGP) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.interp]: Autogenerated file added by error. Removing.

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.tokens (hunk 3 above)
Comment (LuisGP) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDL.tokens]: Autogenerated file added by error. Removing.

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.java (hunk 4 above)
Comment (LuisGP) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.java]: Autogenerated file added by error. Removing.

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.interp (hunk 5 above)
Comment (LuisGP) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLLexer.interp]: Autogenerated file added by error. Removing.

File: src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLParser.java (hunk 6 above)
Comment (LuisGP) [reply src/main/antlr4/omg/com/eprosima/idl/parser/grammar/.antlr/IDLParser.java]: Autogenerated file added by error. Removing.

**Pull request merged**